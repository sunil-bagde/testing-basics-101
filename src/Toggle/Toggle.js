import { Component } from "react";
import "./Toggle.css";

class Toggle extends Component {
    render() {
        const { toggle , onToggle = () => {} } = this.props;
     
        return (
            <span
                className="toggle"
                role="checkbox"
                tabIndex="0"
                onClick={onToggle}
                data-testid='toggle' 
                aria-checked={toggle.toString()}
            ></span>
        );
    }
}

export default Toggle;
