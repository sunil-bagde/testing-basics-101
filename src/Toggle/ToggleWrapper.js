import { useState } from "react";
import Toggle from "./Toggle";

function ToggleWrapper() {
  const [toggle, setToogle] = useState(false);
  return (
    <div
      style={{
        height: "100vh",
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
      }}
    >
      <h2 data-testid="toggle-text">{toggle ? "Yes" : "No"}</h2>
      <Toggle toggle={toggle} onToggle={() => setToogle(!toggle)} />
    </div>
  );
}
export default ToggleWrapper;
