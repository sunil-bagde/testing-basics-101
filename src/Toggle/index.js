import Toggle from "./Toggle";
import ToggleWrapper from "./ToggleWrapper";

export default Toggle;
export { ToggleWrapper };
