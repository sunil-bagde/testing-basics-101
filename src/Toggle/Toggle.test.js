import { shallow, mount } from "enzyme"; 
import toJson from "enzyme-to-json";
import ToggleWrapper from "./ToggleWrapper";
import Toggle from "./Toggle";

import { getByTestId } from "../test/utils";

describe("<ToggleWrapper />", () => {
     test("matches saved snapshot <ToggleWrapper /> ", () => {
        const wrapper = shallow(<ToggleWrapper />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test("should render default state", () => {
        const wrapper = shallow(<ToggleWrapper />);
        const toggleButton = wrapper.find(getByTestId("toggle-text"));
        const checker = toggleButton.text();
        expect(checker).toBe("No");
        wrapper.unmount();
    });

    test("Should changed based on props", () => {
        const wrapper = shallow(<Toggle toggle={false} />);
        const toggleButton = wrapper.find(getByTestId("toggle"));
        const checker = toggleButton.props()["aria-checked"];
        expect(checker).toBe("false");
        wrapper.unmount();
    });

    test("should 'Toggle' on click ", () => {
        const wrapper = mount(<ToggleWrapper />);
        let toggleButton = wrapper.find(getByTestId("toggle"));
        toggleButton.simulate("click"); 
        wrapper.update();
        toggleButton = wrapper.find(getByTestId("toggle"));
        const toggleButton1 = wrapper.find(getByTestId("toggle-text"));
        expect(toggleButton1.text()).toBe("Yes");
        expect(toggleButton.props()["aria-checked"]).toBe("true");
        wrapper.unmount();
    });

});
