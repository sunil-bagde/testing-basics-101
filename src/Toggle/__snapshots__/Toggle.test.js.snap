// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleWrapper /> matches saved snapshot <ToggleWrapper />  1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": "100vh",
      "justifyContent": "center",
    }
  }
>
  <h2
    data-testid="toggle-text"
  >
    No
  </h2>
  <Toggle
    onToggle={[Function]}
    toggle={false}
  />
</div>
`;
